app/front/scripts/services/history.js

```diff
@@ -0,0 +1,47 @@
+/**
```

Comment:
Adding on that, this is actually a big issue - 
since we need a way to open the viewer with a specific dataset showing (e.g. right after we finish loading a dataset)